src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -96,6 +97,7 @@ public function build(ContainerBuilder $container)
         $this->addCompilerPassIfExists($container, TranslatorPass::class);
         $container->addCompilerPass(new LoggingTranslatorPass());
         $container->addCompilerPass(new AddCacheWarmerPass());
+        $container->addCompilerPass(new CachePoolPrunerPass());
```

Comment:
this pass must be registered next to CachePoolClearerPass, as TYPE_AFTER_REMOVING
this is to allow removable of unused (prunebale) cache pool, which the current code actually prevents, by always creating a reference to them for the command